packages/remirror__extension-link/src/link-extension.ts

```diff
@@ -238,6 +272,8 @@ export type LinkAttributes = ProsemirrorAttributes<{
     openLinkOnClick: false,
     autoLinkRegex: DEFAULT_AUTO_LINK_REGEX,
     autoLinkAllowedTLDs: TOP_50_TLDS,
+    findAutoLinks: null,
+    isValidUrl: null,
```

Comment:
I might be wrong here, but I _think_ we can utilise the decorator here to provide default arguments. Rather than doing

```ts
if (this.options.findAutoLinks) {
  return this.options.findAutoLinks(str, this.options.defaultProtocol);
}
```

We can use `this` to provide default arguments, so we could do

```suggestion
    findAutoLinks: this.findAutoLinks,
    isValidUrl: this.isValidUrl,
```

[Example here](https://github.com/remirror/remirror/blob/ac2019fc59b8caf1f1f8eaaa3163b5a15a459c5c/packages/remirror__core/src/builtins/helpers-extension.ts#L159-L168)